### ISSUE TITLE
Benign error: tee: /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages: No such file or directory

### ISSUE THREAD

**Issue by chrispsommers** [ipdk-io/ipdk#7]
Got an apparently benign error after executing `/root/scripts/set_hugepages.sh` per the instructions on https://github.com/ipdk-io/ipdk/blob/main/build/IPDK_Container/README (step 6.a.):
```
[root@aea8dd7ac07e P4-OVS]# source /root/P4-OVS/p4ovs_env_setup.sh /root/p4-sde/install 
OS and Version details...
Fedora : 33


Updated Environment Variables ...
SDE_INSTALL: /root/p4-sde/install
PKG_CONFIG_PATH: /root/p4-sde/install/lib64/pkgconfig
LIBRARY_PATH: 
LD_LIBRARY_PATH: :/root/p4-sde/install/lib:/root/p4-sde/install/lib64:/usr/local/lib:/usr/local/lib64
PATH: /root/P4-OVS/:/root/scripts/:/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
C_INCLUDE_PATH: 
CPLUS_INCLUDE_PATH: 

[root@aea8dd7ac07e P4-OVS]# /root/scripts/set_hugepages.sh
1024
tee: /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages: No such file or directory
1024
```

**Comment by stolsma** [comment]
I executed the set_hugepages.sh file on the host instead of in the container and then it started up the ovs processes! Maybe that works for you too @chrispsommers ?

**Comment by chrispsommers** [comment]
The same error occurs whether I execute it from the host or inside the container.

**Comment by rst0git** [comment]
> tee: /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages: No such file or directory

This error is expected unless you have two numa nodes. (See `man 3 numa`)

**Comment by chrispsommers** [comment]
Can the script be changed to not try something which will fail (benign or not)? Or add comments in the README to advise users not to worry. Thanks.

**Comment by stolsma** [comment]
Can't the script test if numa nodes exist and if they do execute the command for each numa node?

**Comment by asafravid** [comment]
I got to here as well, same issue

    Updated Environment Variables ...
    SDE_INSTALL: /root/p4-sde/install
    PKG_CONFIG_PATH: /root/p4-sde/install/lib/x86_64-linux-gnu/pkgconfig
    LIBRARY_PATH:
    LD_LIBRARY_PATH: :/root/p4-sde/install/lib:/root/p4-sde/install/lib/x86_64-linux-gnu:/usr/local/lib:/usr/local/lib64
    PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    C_INCLUDE_PATH:
    CPLUS_INCLUDE_PATH:

    root@05eb99eb621b:~/P4-OVS# /root/scripts/set_hugepages.sh
    1024
    tee: /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages: No such file or directory
    1024
    root@05eb99eb621b:~/P4-OVS#

Does this mean I need a specific server (with 2 Numa nodes)? Or can this be considered as a "Warning" only and I can continue with the Readme steps?

**Comment by chrispsommers** [comment]
It didn't seem to affect operation, just a warning. That's what I meant by "benign."

**Comment by asafravid** [comment]
@chrispsommers understood, thanks for the clarification

**Comment by asafravid** [comment]
@chrispsommers as far as I recall I’ve followed each and every step in the readme but I’ll double check and update

**Comment by mestery** [comment]
This should be fixed as a part of #16, I'll close this for now and someone can open a new one if they run into huge pages issues in the future.
